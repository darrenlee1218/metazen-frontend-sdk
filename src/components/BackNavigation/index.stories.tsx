import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import BackNavigation from '.';

export default {
  title: 'common-components/stateless/molecules/BackNavigation',
  component: BackNavigation,
  argTypes: {},
} as ComponentMeta<typeof BackNavigation>;

const Template: ComponentStory<typeof BackNavigation> = (args) => <BackNavigation {...args} />;

export const Default = Template.bind({});

Default.args = {
  label: 'Matic',
};
