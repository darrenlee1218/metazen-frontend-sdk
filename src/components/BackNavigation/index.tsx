import React from 'react';

import Box from '@mui/material/Box';
import { CSSObject } from '@mui/material/styles';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';

interface BackNavigationProps {
  /**
   * label for the sub-navbar
   */
  label?: string;

  /**
   * optional - custom back button press event
   */
  onBackPress: () => void;
  sx?: CSSObject;
  labelProps?: CSSObject;
}

export const BackNavigation: React.FunctionComponent<BackNavigationProps> = ({
  label,
  onBackPress,
  sx = {},
  labelProps = {},
}) => {
  return (
    <Box
      sx={{
        mx: '16px',
        my: '12px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        ...sx,
      }}
    >
      <IconButton size="small" onClick={onBackPress}>
        <ArrowBackIcon sx={{ fontSize: '16px' }} />
      </IconButton>
      <Typography
        align="center"
        variant="h2"
        color="text.primary"
        sx={{ fontSize: { lg: 20, md: 18, sm: 16 }, fontWeight: 600, ...labelProps }}
      >
        {label}
      </Typography>
      {/* Empty box for justify-content */}
      {/* make sure keep same with as IconButton */}
      <Box sx={{ width: '26px' }} />
    </Box>
  );
};

export default BackNavigation;
