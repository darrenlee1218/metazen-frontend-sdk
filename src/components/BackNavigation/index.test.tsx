import React from 'react';
import TestRenderer from 'react-test-renderer';

import BackNavigation from './index';

describe('Test snapshot BackNavigation components', () => {
  it('should be render correctly', () => {
    const tree = TestRenderer.create(<BackNavigation label="Home" onBackPress={() => {}} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
