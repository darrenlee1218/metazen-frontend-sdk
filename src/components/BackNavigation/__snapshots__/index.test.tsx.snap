// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test snapshot BackNavigation components should be render correctly 1`] = `
<div
  className="MuiBox-root css-tfe0o3"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-xtuuzo-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
  </button>
  <h2
    className="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter css-1rn45oi-MuiTypography-root"
  >
    Home
  </h2>
  <div
    className="MuiBox-root css-8h2w1h"
  />
</div>
`;
